irohad/maintenance/metrics.hpp

```diff
@@ -8,9 +8,40 @@
 
 #include <string>
 #include <memory>
+#include <optional>
+#include <thread>
 #include <prometheus/registry.h>
+#include <prometheus/exposer.h>
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/iroha_internal/block.hpp"
+#include "main/subscription.hpp"
+#include "network/ordering_gate_common.hpp"
+#include "ametsuchi/wsv_query.hpp"
+#include "ametsuchi/storage.hpp"
+#include "logger/logger_fwd.hpp"
 
-std::shared_ptr<prometheus::Registry>
-    maintenance_metrics_init(std::string const& listen_addr);
+class Metrics : public std::enable_shared_from_this<Metrics> {
+  using OnProposalSubscription = iroha::BaseSubscriber<
+      bool,iroha::network::OrderingEvent>;  //FixMe subscribtion ≠ subscriber
```

Comment:
Either remove or comment, or please create a gihub issue for that

Reply:
@iceseer Would you like to rename some classes to fix terminalogy.